src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -136,10 +152,49 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
   closePopover() {
     this.setState({ ...this.state, isPopoverOpen: false });
   }
+  getFilteredDefaultDataSource(defaultDataSource: string | null): string | null {
+    if (!defaultDataSource) return defaultDataSource;
+    const { dataSourceFilter } = this.props;
+    // check if the defaultDataSouce can be filtered out
+    const filteredDataSource = dataSourceFilter
+      ? [{ id: defaultDataSource }].filter(dataSourceFilter)
+      : [{ id: defaultDataSource }];
+    return filteredDataSource.length === 0 ? null : filteredDataSource[0].id;
+  }
+  /**
+   * when call handleSwitchDefaultDatasource, the default data source must exist,
+   * since the button only display when defaultDataSource exist
+   */
+  async handleSwitchDefaultDatasource(defaultDataSourceId: string | null) {
```

Comment:
This function can take null and might cause error when calling getDataSourceById when the input id is null, however, when we use this function, the input dataSourceId always exist as string